nml/actions/action2layout.py

```diff
@@ -287,13 +287,18 @@ def _validate_sprite(self, name, value):
             self.sprite_from_action1 = True
             val, offset = self.resolve_spritegroup_ref(value)
             if offset is not None:
-                self.create_register(name, offset)
+                if offset.supported_by_actionD(False):
+                    val = expression.BinOp(nmlop.ADD, val, offset, value.pos)
+                else:
+                    self.create_register(name, offset)
```

Comment:
Not sure about this change, I think it will break things because `1 | 0002 | Add offset to sprite, disable default usage of construction stage or railtype-offset` flag is not set if actionD is used

Reply:
After more reading of the spec I decided it was definitely wrong.